### ISSUE TITLE
All SO(3)/SE(3)/SE(2) exp and log return nan gradients at the identity

### ISSUE THREAD

**Issue by mohansree14** [kornia/kornia#4404]
## Summary

Every `exp` and `log` in `kornia.geometry.liegroup` returns **`nan` gradients at the identity**, on every dtype
including `float64`. The forward values are correct; only the backward pass is poisoned. The zero rotation vector
is the standard initialisation for pose optimisation, so this is the first backward pass of a typical
`So3.exp(...)`-based optimisation loop.

`So2` is the exception — it goes through `atan2` and has no singular branch.

| | forward at identity | backward at identity |
| --- | --- | --- |
| `So3.exp` | correct | **`nan`** |
| `So3.log` | correct | **`nan`** (also at a half turn, `real = 0`) |
| `Se3.exp` | correct | **`nan`** |
| `Se3.log` | correct | **`nan`** |
| `Se2.exp` | correct | **`nan`** |
| `So2.exp` / `So2.log` | correct | finite |

```python
import torch
from kornia.geometry.liegroup import So3, Se3, Se2
from kornia.geometry.quaternion import Quaternion

for dt in (torch.float64, torch.float32, torch.float16):
    v = torch.zeros(1, 3, dtype=dt, requires_grad=True)
    So3.exp(v).matrix().sum().backward()
    print(dt, "So3.exp", v.grad.flatten().tolist())
```

```
torch.float64 So3.exp [nan, nan, nan]
torch.float32 So3.exp [nan, nan, nan]
torch.float16 So3.exp [nan, nan, nan]
```

Same shape for the other four:

```python
q = torch.tensor([[1., 0., 0., 0.]], dtype=torch.float64, requires_grad=True)
So3(Quaternion(q)).log().sum().backward()            # [nan, nan, nan, nan]

q = torch.tensor([[0., 1., 0., 0.]], dtype=torch.float64, requires_grad=True)
So3(Quaternion(q)).log().sum().backward()            # [nan, nan, nan, nan]  (half turn)

v = torch.zeros(1, 6, dtype=torch.float64, requires_grad=True)
Se3.exp(v).matrix().sum().backward()                 # [nan] * 6

v = torch.zeros(1, 3, dtype=torch.float64, requires_grad=True)
Se2.exp(v).matrix().sum().backward()                 # [0., 0., nan]
```

## Mechanism

`torch.where` differentiates **both** branches and multiplies the unselected one by zero. Where that branch's
value is a `0/0` or its derivative is unbounded, `0 * nan` and `0 * inf` are `nan`, and the poison reaches the
input even though the value came from the safe branch. This is the same class as the merged #4338
(`solve_cubic`'s `D > 0` branch poisoning other rows) and #4233.

Per site:

- `kornia/geometry/liegroup/so3.py:132` — `b_large = torch.sin(theta_half) / theta` is `0/0` at `theta = 0`,
  and `b = torch.where(small_mask, b_small, b_large)` walks it anyway.
- `kornia/geometry/liegroup/so3.py:150-156` — three singularities in one expression: `sqrt` at `vec_sq = 0`,
  the division by `theta`, and `acos(real)` at `real = ±1`, all in the branch that is *not* selected at the
  identity; symmetrically, the small-angle branch `2 * vec / real` divides by zero at a half turn, which is the
  branch not selected there.
- `kornia/geometry/liegroup/se3.py:180-187` — `theta = dot(omega, omega).sqrt()` has an unbounded derivative at
  `omega = 0`, and `V` divides by `theta**2` and `theta**3`.
- `kornia/geometry/liegroup/se3.py:201` — `clamp_min(1e-12).sqrt()`. Per #4229, `clamp` is not a gradient guard
  on torch < 2.14 (it passes the incoming gradient straight through at its bound), and `1e-12` underflows to `0`
  in `float16` regardless, so `sqrt`'s boundary derivative is reached on every supported version.
- `kornia/geometry/liegroup/se2.py:191-192` — `so2.z.imag / theta` and `(1 - so2.z.real) / theta` are `0/0` at
  `theta = 0`.

## Why it was not caught

`test_gradcheck` is an empty `# TODO: implement me` stub in all three test classes —
`tests/geometry/liegroup/test_so3.py:50`, `tests/geometry/liegroup/test_se3.py:68`,
`tests/geometry/liegroup/test_se2.py:97`. Nothing in the suite differentiates these functions anywhere, let
alone at the identity.

## Suggested fix

The version-independent pattern already accepted in #4233, #4338 and #4228: substitute a safe argument into the
branch that gets differentiated, so the singular expression is never evaluated at its singular point. The
substituted value is discarded by the `where`, so every forward stays byte-identical.

I have this working locally for all five functions — forwards byte-identical across 64 random inputs per
function plus small-angle and sub-`eps` batches, `gradcheck` passing at generic points, and the identity
gradients agreeing with central differences (for `So3.log` at the identity, with the differences taken along the
unit sphere rather than through the ambient 4-space, where the perturbed quaternion leaves the manifold and the
other branch's formula takes over). Happy to open the PR, along with real `test_gradcheck` implementations for
the three stubs.

## Environment

kornia `main` @ `d6a2c13`, torch 2.7.0, python 3.11, Windows, CPU.
Related: #4229 (`clamp` is not a portable gradient guard), #4338 and #4233 (same `torch.where` poisoning class),
#4007 / #4228 (the same defect in `quaternion_exp_to_log`).
